Make `Debug` representations of `[Lazy, Once]*[Cell, Lock]` consistent with `Mutex` and `RwLock`

joboet opened the pull request:
`Mutex` prints `<locked>` as a field value when its inner value cannot be accessed, but the lazy types print a fixed string like "`OnceCell(Uninit)`". This could cause confusion if the inner type is a unit type named `Uninit` and does not respect the pretty-printing flag. With this change, the format message is now "`OnceCell(<uninit>)`", consistent with `Mutex`.

Mark-Simulacrum commented on library/std/src/sync/mutex.rs:
```diff
@@ -480,13 +480,7 @@ impl<T: ?Sized + fmt::Debug> fmt::Debug for Mutex<T> {
-                    }
-                }
-                d.field("data", &LockedPlaceholder);
+                d.field("data", &fmt::DebugPlaceholder("<locked>"));
```
I think we can achieve equivalent results in all of these cases with `format_args!("<locked>")` as the thing passed in, which avoids the extra type and works on stable.

joboet replied:
Oh, yeah, that works!

joboet replied:
I've changed it now.

joboet committed:
std: make `Debug` representations of `[Lazy, Once]*[Cell, Lock]` consistent with `Mutex` and `RwLock`

`Mutex` prints `<locked>` as a field value when its inner value cannot be accessed, but the lazy types print a fixed string like "`OnceCell(Uninit)`". This could cause confusion if the inner type is a unit type named `Uninit` and does not respect the pretty-printing flag. With this change, the format message is now "`OnceCell(<uninit>)`", consistent with `Mutex`.

bors commented:
:umbrella: The latest upstream changes (presumably #109483) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

workingjubilee commented:
r? libs-api

Please FCP (or, if this is a trivial enough change, decide it doesn't need FCP). This is a simple change that makes the Debug formats more consistent but it _is_ user-facing.

workingjubilee commented:
For the record, I am in favor.

dtolnay reviewed (approved):
Thanks!

dtolnay commented:
@bors r+

bors commented:
:pushpin: Commit b7e68dfc945c4cf8d1753cc3a8be48a71bb39d5a has been approved by `dtolnay`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=dtolnay b7e68dfc945c4cf8d1753cc3a8be48a71bb39d5a -->
<!-- homu: {"type":"Approved","sha":"b7e68dfc945c4cf8d1753cc3a8be48a71bb39d5a","approver":"dtolnay","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
